### PR TITLE
support generate railroad diagram for sql grammar of mysql

### PR DESCRIPTION
**Pull request by tianhao960**
Ref #21616.

Changes proposed in this pull request:
  - support generate railroad diagram for sql grammar of mysql
image

the default output dir is ${project.build.directory}/generated-sources/rr
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: sql-parser/dialect/mysql/pom.xml [hunk 1]
```diff
@@ -46,6 +46,22 @@
...
+                        <goals>
+                            <goal>rr</goal>
+                        </goals>
+                        <configuration>
+                            <rootRule>execute</rootRule>
+                            <libDirectory>src/main/antlr4/imports/mysql/</libDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (TeslaCN) [inline sql-parser/dialect/mysql/pom.xml:64]: Hi @tianhao960 
Is there any detail of this plugin?

**Review by TeslaCN — CHANGES_REQUESTED** [review]
I found nothing about this plugin from Google and Maven repository. We have to make sure we are using a license-compatible and safe open source plugin.

File: sql-parser/dialect/mysql/pom.xml (hunk 1 above)
Comment (tianhao960) [reply sql-parser/dialect/mysql/pom.xml:64]: https://github.com/tianhao960/rrg-antlr4

**Comment by tianhao960** [comment]
provide more license info about the plugin:

https://github.com/tianhao960/rrg-antlr4
thanks.

**Comment by mlycore** [comment]
@tianhao960 issue #21616 has some latested comments, please check it out.

**Comment by strongduanmu** [comment]
Since this pr has long time no response, I will close it.

**Pull request closed without merging**